core/drivers/crypto/caam/cipher/local.h

```diff
@@ -0,0 +1,72 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   CAAM Cipher Local header.
+ */
+#ifndef __LOCAL_H__
+#define __LOCAL_H__
+
+#include <drvcrypt.h>
+#include <drvcrypt_cipher.h>
+
+/*
+ * Definition of flags tagging which key(s) is required
+ */
+#define NEED_KEY1  BIT(0)
+#define NEED_KEY2  BIT(1)
+#define NEED_IV    BIT(2)
+#define NEED_TWEAK BIT(3)
+
+/*
+ * Cipher Algorithm definition
+ */
+struct cipheralg {
+	uint32_t type;             /* Algo type for operation */
+	uint8_t size_block;        /* Computing block size */
+	uint8_t size_ctx;          /* CAAM Context Register size */
+	uint8_t ctx_offset;        /* CAAM Context Register offset */
+	uint8_t require_key;       /* Tag defining key(s) required */
+	struct caamdefkey def_key; /* Key size accepted */
+
+	TEE_Result (*update)(struct drvcrypt_cipher_update *dupdate);
```

Comment:
`dupdate`?

Reply:
`d` for `drvcrypt` ? Are you ok with this naming?

Reply:
Yeah, it's OK.